Readme.mkd

```diff
@@ -97,19 +97,27 @@ class Duck < ActiveRecord::Base
   include RankedModel
 
   ranks :row_order,                  # Name this ranker, used with rank()
-    :column => :sort_order           # Override the default column, which defaults to the name
+    column: :sort_order              # Override the default column, which defaults to the name
 
   belongs_to :pond
   ranks :swimming_order,
-    :with_same => :pond_id           # Ducks belong_to Ponds, make the ranker scoped to one pond
+    with_same: :pond_id              # Ducks belong_to Ponds, make the ranker scoped to one pond
 
   ranks :row_order,
-    :with_same => [:pond_id, :breed] # Lets rank them by breed
+    with_same: [:pond_id, :breed]    # Lets rank them by breed
 
-  scope :walking, where(:walking => true )
+  scope :walking, where(walking: true )
   ranks :walking_order,
-    :scope => :walking               # Narrow this ranker to a scope
+    scope: :walking                  # Narrow this ranker to a scope
 
+  belongs_to :parent, class_name: 'Duck', optional: true
+  ranks :child_order,
+    unless: :parent?,                # Rank only ducks that have a parent. Alternatively a Proc or lambda can be passed, e.g. proc { parent.nil? }
```

Comment:
Hi @aried3r, this is a good case for adding an `:if` in addition to `:unless`. The name of your method in this example is confusing though the logic stacks up. It should probably be:

```
def nil_parent?
  parent.nil?
end
```

You could also just check for `!parent`.

Would you be interested in adding an `:if` option to this PR and documenting that too? The documentation for `:unless` should probably be changed to something that is more commonplace.

```
ranks :child_order, with_same: :parent_id, if: :parent
```

No need to a supporting method in this case as `parent` returns `nil` if the relation doesn't exist.

Reply:
Thank you for your review!

> The name of your method in this example is confusing though the logic stacks up. It should probably be:
> 
> ```
> def nil_parent?
>   parent.nil?
> end
> ```

I agree, but every `unless` can be turned into an `if`, and the point is to show an `unless` example 😅. I read it as "rank ducks, unless the duck is a parent". How to check if a duck is a parent? `duck.parent?`, since the Ruby style is usually to avoid `is_` prefixes. 

> No need to a supporting method in this case as parent returns nil if the relation doesn't exist.

I am aware, but it just serves as documentation on how to use supporting methods if you don't want to use `proc` or `->`. I guess we could make it more general like:

```Ruby
ranks :child_order, with_same: :parent_id, if: :some_complicated_check

def some_complicated_check
  # Your business logic here
end
```

> The documentation for `:unless` should probably be changed to something that is more commonplace.
> 
> ```
> ranks :child_order, with_same: :parent_id, if: :parent
> ```


I'm open to that! Can you think of an example where `unless` would make more sense? As for the example, I believe that would be more confusing. What the code does is "rank children with the same `parent_id` if a parent is present", but the code rather reads "rank children with the same `parent_id` if ? parent" where "?" is of course "[if] duck has a [parent]", but it might also be read as "[if] duck is [a parent]". That's of course making a lot of assumptions.

Maybe I can think of something better, or we go for the general case mentioned above? WDYT?

Reply:
That being said, I can add an `if` option in a separate PR. Might be easier once we figure out what is left to do here so I can pretty much copy over examples etc.

Reply:
Hehe, yes I see where the confusion was (is a vs has a). Maybe renaming the method?

```
ranks :child_order, with_same: :parent_id, unless: :has_no_parent?

def has_no_parent?
  parent.nil?
end
```

Would that solve this PR? Sorry for the unneeded coding lesson in my original comment, I didn't get the cue that the documentation example was contrived :D Have to turn off my optimisation brain sometimes!

Reply:
> Sorry for the unneeded coding lesson in my original comment, I didn't get the cue that the documentation example was contrived :D Have to turn off my optimisation brain sometimes!

No worries, I appreciate a tough review. :)

I pushed my latest changes according with your suggestions.